src/common/mfu_daos.c

```diff
@@ -1313,3 +1315,120 @@ int mfu_flist_copy_daos(
+
+        /* set other args based on if hdf5 cont is src or dst */
+        if (mfu_src_file->type == HDF5 && mfu_dst_file->type == HDF5) {
+            /* TODO: do not use h5repack unless moving
+             * HDF5 containers to/from POSIX (it does not work),
+             * for this case we should possibly use the cont clone
+             * and make sure to set the cont property to HDF5
+             * when the destination is created. For now, error
+             * out and report that this is not yet supported
+             */
+            MFU_LOG(MFU_LOG_ERR, "Copying HDF5 containers where "
+                    "DAOS is the source and destination is not "
+                    "yet supported" MFU_ERRF, MFU_ERRP(-MFU_ERR_INVAL_ARG));
+        } else if (mfu_src_file->type == HDF5) {
+            args[1] = "--dst-vol-name=native";
+            args[2] = strdup(src_cont_uuid);
+            args[3] = strdup(argpaths[1]);
+            args[4] = NULL; 
+        } else if (mfu_dst_file->type == HDF5) {
+            args[1] = "--src-vol-name=native";
+            args[2] = strdup(argpaths[0]);
+            args[3] = strdup(dst_cont_uuid);
+            args[4] = NULL; 
+        } else {
+            MFU_LOG(MFU_LOG_ERR, "Error parsing DAOS HDF5 arguments"
+                    MFU_ERRF, MFU_ERRP(-MFU_ERR_INVAL_ARG));
+            mfu_free(&args[3]);
+            rc = 1;
+            return rc;
+        }
+
+        /* Child process */
+        if ((child = fork()) == 0) {
+
+            /* start h5repack */
+            execvp(args[0], args);
+
+            /* if child process reaches this point then execvp must have failed */
+            MFU_LOG(MFU_LOG_ERR, "execvp on h5repack failed: "
+                    MFU_ERRF, MFU_ERRP(-MFU_ERR_DCP));
+            mfu_free(&args[3]);
+            rc = 1;
+            return rc;
+        /* Parent process */
+        } else {
+            if (child == (pid_t)(-1)) {
+                MFU_LOG(MFU_LOG_ERR, "fork failed: "
+                        MFU_ERRF, MFU_ERRP(-MFU_ERR_DCP));
+                mfu_free(&args[3]);
+                rc = 1;
+                return rc;
+            } else {
+                /* parent will wait for child to complete */
+                child_pid = wait(&child_status);
+            }
+        }
+
+        /* if child status is 0, then copy was successful */
+        if (child_status != 0) {
+            MFU_LOG(MFU_LOG_ERR, "errors occurred while copying to/from container: "
+                    MFU_ERRF, MFU_ERRP(-MFU_ERR_DCP));
+            mfu_free(&args[3]);
+            rc = 1;
+            return rc;
+        } else {
+            if (dst_cont_uuid[0] != '\0') {
+                MFU_LOG(MFU_LOG_INFO, "Successfully copied to DAOS HDF5 container: %s",
+                        dst_cont_uuid);
+            } else {
+                MFU_LOG(MFU_LOG_INFO, "Successfully copied to HDF5 file: %s",
+                        argpaths[1]);
+            }
```

Comment:
@dsikich I think this is more explicit. What do you think?
```suggestion
            if (mfu_dst_file->type == HDF5) {
                MFU_LOG(MFU_LOG_INFO, "Successfully copied to DAOS HDF5 container: %s",
                        dst_cont_uuid);
            } else {
                MFU_LOG(MFU_LOG_INFO, "Successfully copied to HDF5 file: %s",
                        argpaths[1]);
            }
```